### PR TITLE
Initialize genesis state at node startup

### PR DESCRIPTION
**Pull request by omerfirmak**

### CODE REVIEW

File: blockchain/blockchain.go [hunk 1]
```diff
@@ -776,11 +776,6 @@ func (b *Blockchain) HeadState() (core.StateReader, StateCloser, error) {
 		return nil, nil, err
 	}
 
-	_, err = chainHeight(txn)
```
Comment (joshklop) [inline blockchain/blockchain.go:779]: Is this change related to adding `InitGenesisState`?

File: blockchain/blockchain.go [hunk 2]
```diff
@@ -1149,3 +1144,51 @@ func (b *Blockchain) Finalise(pending *Pending, sign BlockSignFunc) error {
...
+
+func (b *Blockchain) InitGenesisState(stateDiff *core.StateDiff, classes map[felt.Felt]core.Class) error {
+	return b.database.Update(func(txn db.Transaction) error {
+		_, err := genesisState(txn)
+		if !errors.Is(err, db.ErrKeyNotFound) {
+			return errors.New("genesis state already initiliazed")
+		}
+
+		state := core.NewState(txn)
+		if err = state.Update(0, stateDiff, classes); err != nil {
```
Comment (joshklop) [inline blockchain/blockchain.go:1173]: This will log changes to history for block zero, correct? Do we want that?

File: node/genesis.go [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+
+import (
+	"github.com/NethermindEth/juno/blockchain"
+	"github.com/NethermindEth/juno/genesis"
+	"github.com/NethermindEth/juno/vm"
+)
+
+func buildGenesis(genesisPath string, bc *blockchain.Blockchain, v vm.VM) error {
+	_, err := bc.GenesisState()
+	if err == nil {
```
Comment (joshklop) [inline node/genesis.go:11]: Would this be a bit safer?
```go
if _, err := bc.GenesisState(); !errors.Is(err, db.ErrNotFound) {
       return err
}
```

File: blockchain/blockchain.go (hunk 1 above)
Comment (joshklop) [reply blockchain/blockchain.go:779]: Ah I see. Now we can have a state without a chain.

File: node/genesis.go (hunk 3 above)
Comment (omerfirmak) [reply node/genesis.go:11]: Nice, done.

File: blockchain/blockchain.go (hunk 2 above)
Comment (omerfirmak) [reply blockchain/blockchain.go:1173]: Yeah, it is not ideal, but doesn't really hurt either. Since it requires some changes to `core.State` API, I opted to deal with it later if needed.

**Comment by omerfirmak** [comment]
Merging assuming these were all the comment @joshklop had, we can revisit any part of this PR again as needed.
